src/test/regress/sql/citus_local_tables.sql

```diff
@@ -0,0 +1,174 @@
+
+-- partitiond table tests --
+
+CREATE TABLE partitioned_table(a int, b int) PARTITION BY RANGE (a);
+CREATE TABLE partitioned_table_1 PARTITION OF partitioned_table FOR VALUES FROM (0) TO (10);
+CREATE TABLE partitioned_table_2 PARTITION OF partitioned_table FOR VALUES FROM (10) TO (20);
+
+-- cannot create partitioned citus local tables
+SELECT create_citus_local_table('partitioned_table');
+
+-- cannot create citus local table as a partition of a local table
+BEGIN;
+  CREATE TABLE citus_local_table PARTITION OF partitioned_table FOR VALUES FROM (20) TO (30);
+
+  -- this should fail
+  SELECT create_citus_local_table('citus_local_table');
+ROLLBACK;
+
+-- cannot create citus local table as a partition of a local table
+-- via ALTER TABLE commands as well
+BEGIN;
+  CREATE TABLE citus_local_table (a int, b int);
+
+  SELECT create_citus_local_table('citus_local_table');
+
+  -- this should fail
+  ALTER TABLE partitioned_table ATTACH PARTITION citus_local_table FOR VALUES FROM (20) TO (30);
+ROLLBACK;
+
+-- cannot attach citus local table to a partitioned distributed table
+BEGIN;
+  SELECT create_distributed_table('partitioned_table', 'a');
+
+  CREATE TABLE citus_local_table (a int, b int);
+  SELECT create_citus_local_table('citus_local_table');
+
+  -- this should fail
+  ALTER TABLE partitioned_table ATTACH PARTITION citus_local_table FOR VALUES FROM (20) TO (30);
+ROLLBACK;
+
+-- drop them for next tests
+DROP TABLE citus_local_table_1, citus_local_table_2, distributed_table;
+
+-- create test tables
+CREATE TABLE citus_local_table_1 (a int primary key);
+SELECT create_citus_local_table('citus_local_table_1');
+
+CREATE TABLE citus_local_table_2 (a int primary key);
+SELECT create_citus_local_table('citus_local_table_2');
+
+CREATE TABLE local_table (a int primary key);
+
+CREATE TABLE distributed_table (a int primary key);
+SELECT create_distributed_table('distributed_table', 'a');
+
+CREATE TABLE reference_table (a int primary key);
+SELECT create_reference_table('reference_table');
+
+-- show that colociation of citus local tables are not supported for now
+
+-- between citus local tables
+SELECT mark_tables_colocated('citus_local_table_1', ARRAY['citus_local_table_2']);
+
+-- between citus local tables and reference tables
+SELECT mark_tables_colocated('citus_local_table_1', ARRAY['reference_table']);
+SELECT mark_tables_colocated('reference_table', ARRAY['citus_local_table_1']);
+
+-- between citus local tables and distributed tables
+SELECT mark_tables_colocated('citus_local_table_1', ARRAY['distributed_table']);
+SELECT mark_tables_colocated('distributed_table', ARRAY['citus_local_table_1']);
+
+-----------------------------------
+---- utility command execution ----
+-----------------------------------
+
+-- any foreign key between citus local tables and other tables cannot be set for now
+-- most should error out (for now with meaningless error messages)
+
+-- between citus local tables
```

Comment:
We can also have tests where a table references a some other table before creating the citus local table.

Reply:
As now we don't support initial fkeys with other tables, I will add some tests with self referencing fkeys

Reply:
added